src/collective/cover/tests/test_collection_tile.py

```diff
@@ -164,6 +164,26 @@ def test_offset(self):
         self.assertEqual(len(items), 1)
         self.assertEqual(items[0].getId(), 'my-image1')
 
+    def test_random_items(self):
+        obj = self.portal['mandelbrot-set']
+        self.tile.populate_with_object(obj)
+
+        items_not_random = [r for r in self.tile.results()]
+        self.assertEqual(
+            [r for r in self.tile.results()],
+            items_not_random
+        )
+
+        self.tile.data['random'] = True
+        random_items1 = [r for r in self.tile.results()]
+        random_items2 = [r for r in self.tile.results()]
+        random_items3 = [r for r in self.tile.results()]
+        self.assertTrue(
```

Comment:
@rodfersou you must avoid most logic on tests; also, there is a chance that this fails randomly.